### PR TITLE
Prefix any C-exposed symbols with `bs_`

### LINKED ISSUE #53: Prefix C-level functions to avoid collisions
This would only affect users who directly include `bridgestan.h` in a project, but it is useful (and [fairly standard](https://halt.software/libraries-in-c-prefix-or-not/), I believe) to prefix C function names with a common string to avoid namespace issues that can arise in C. So, our C API would have functions like `bs_name`, while Python/Julia/R etc would still just use `name`.

I'm happy to make these changes, the question is just which prefix to use:

- `bs` short, but commonly used for another meaning
- `stan` - still short, `stan_log_density` etc all read naturally, but this isn't an official Stan project
- `bridgestan` - most accurate and unambiguous, but very long

### PR DESCRIPTION
**Pull request by WardBrian**
Closes #53.

This is a breaking change for the C API, but not for any of the higher level APIs.

### CODE REVIEW

File: julia/src/model.jl [hunk 1]
```diff
@@ -57,7 +59,7 @@ mutable struct StanModel
 
         function f(sm)
             ccall(
-                Libc.Libdl.dlsym(sm.lib, "destruct"),
+                Libc.Libdl.dlsym(sm.lib, "bs_destruct"),
```
Comment (bob-carpenter) [inline julia/src/model.jl:62]: `construct` -> `bs_construct` (line 47).

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
I only found one place where it looks like the names weren't changed consistently.  I opened all the files and checked everything was changed to `bs_` as appropriate.  Thanks for doing this!

File: julia/src/model.jl (hunk 1 above)
Comment (WardBrian) [reply julia/src/model.jl:62]: Good catch, I did a search for the string `sm.lib` but in the constructor itself the library is called only `lib`. I assume the tests would have caught it rather quickly, but I only ran the Python ones locally before pushing

**Review by bob-carpenter — APPROVED** [review]
looks good.  and you found a missing one in R that I missed.

**Pull request merged**